### DISCUSSION TITLE
Clicking taiko difficulties on beatmap card will put you in taiko mode, and does not revert upon clicking to a different beatmap

### DISCUSSION THREAD

**Discussion by Sublimelime** [ppy/osu#17136] [Ideas] [answered]
## The issue

Just based off the (kinda long, sorry) title this might not seem like unintended behavior - the issue is when you *accidentally* click the taiko (+mania?) difficulties while you are in osu!standard mode (which is easy to do if you click the right part of the beatmap card while just trying to scroll/select it) - it puts you in taiko mode, but doesn't revert upon clicking to the next beatmap or even back to the standard difficulties (which is what I would expect personally, since it should only put me in taiko mode for those difficulties that are specific to taiko, same for mania-only diffs).

Once it's done this, you have to A) notice it's done so (it plays the taiko sound effect upon switching but it's sometimes easy to miss with the song playing) and B) go and put it back in osu!standard mode. Failing to notice can make you accidentally launch a taiko play when you didn't mean to, which is a minor/medium annoyance.

A video of the problem:

https://streamable.com/lgmjr8

## Proposed fix(es):

- Not show taiko (or any other gamemode's) difficulties while not in that gamemode's mode. (Could be an option in settings)
- Make the mode-switch revert upon clicking into a different beatmap set or even back into non-specific-to-taiko diffs. Obviously always revert back to whatever mode is manually selected at the top left (as this problem also affects mania/catch players).

**Comment by bdach** [comment, marked as answer]
duplicate of https://github.com/ppy/osu/issues/9017 / https://github.com/ppy/osu/issues/9032
